### PR TITLE
testsuite: ztest: Add zassert_within macro

### PR DESCRIPTION
**Pull request by nordic-krch**
Added macro to test value against another value with
accepted delta.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
@nashif can you add other reviewers, i don't see ztest maintainer in CODEOWNERS

File: subsys/testsuite/ztest/include/ztest_assert.h [hunk 1]
```diff
@@ -168,6 +168,19 @@ static inline void z_zassert(int cond,
 	zassert((void *)(a) == (void *)(b), #a " not equal to  " #b, \
 		msg, ##__VA_ARGS__)
 
+/**
+ * @brief Assert that @a is within @a b with delta @a d
```
Comment (anangl) [inline subsys/testsuite/ztest/include/ztest_assert.h]: `a` missing after the `@a` tag.

File: subsys/testsuite/ztest/include/ztest_assert.h [hunk 2]
```diff
@@ -168,6 +168,19 @@ static inline void z_zassert(int cond,
...
+ *
+ * @param a Value to compare
+ * @param b Value to compare
+ * @param d Delta
+ * @param msg Optional message to print if the assertion fails
+ */
+#define zassert_within(a, b, d, msg, ...)			     \
+	zassert(((a) > ((b) - (d))) && ((a) < ((b) + (d))),	     \
+		#a " not within  " #b " +/- delta " #d,		     \
```
Comment (anangl) [inline subsys/testsuite/ztest/include/ztest_assert.h]: Double space after "within". Intentional?
Perhaps the word "delta" could be omitted in the message.

**Comment by nordic-krch** [comment]
@anangl thanks for the review. I've applied your findings.

**Review by anangl — APPROVED** [review]
The last thing I'm wondering about is if we should mention that no typecasting is done in this macro. Hence, the comparison is vulnerable to over- and underflows.
For instance, for `u8_t a = 0, b = 250, d = 10;` the assert would pass.

**Pull request merged**